### PR TITLE
New gp 766

### PR DESCRIPTION
**Pull request by h4d4**
This PR updates deployments scripts to include new general propose backend(gp) that will be called vocms0766

- Regarding commit '[648f895](https://github.com/dmwm/deployment/commit/648f895c685c469cdeec47dd25cd37e74aafb135) '  below is the list of `services*/deployment` where general propose backends are `'disable'`, therefore on those scripts the new gp need to be included:

```
#gps ARE LISTED TO BE 'disable' =>  below services are NOT DEPLOYED in gps nodes
confdb
couchdb
crabcache
das
dcafpilot (no deployed anymore in the cluster)
dqmgui
filemover
mongodb
overview (no deployed anymore in the cluster)
phedexreplicamonitoring
reqmgr2ms (until now just in testbed nodes)
tfaas
wmarchive (##updating their deployment script even when this is not deployed into cluster)
```
In that commit I've also included `exporters/manage`, the script made a explicity list of nodes where exporters runs, and there gps are listed.

-  I've not modified below  `services*/deployment` scripts due to gps are not listed to be  'disable', it means that those services are deployed into gps, then is not necessary update it:
```
###SERVICES WHERE GPS ARE no LISTED TO BE 'disable' => below services are DEPLOYED in GPS nodes ##deployment updates, is not necessary
crabserver
dbs
dbsmigration
dbsweb (no deployed anymore in the cluster)
dmwmmon
exporters
gitweb (no deployed anymore in the cluster)
phedex
popdbweb
reqmgr2 
reqmon
sitedb
t0_reqmon = CMS T0 WMStats (T0ReqMon)
t0wmadatasvc
victorweb
```
`NOTE: ` 
In the end, what decides if x deployed service process queries or no, are `fronted-rules` [(backends-prod.txt)](https://github.com/dmwm/deployment/blob/master/frontend/backends-prod.txt). So, there are cases where one service is deployed in several cmsweb's nodes but service's requests are redirected to certain of those nodes  by `fronted-rules`  
Looks like that configuration of deploy services in nodes where finally request will no be handled is part of the way in which the cluster was designed.

- I've not included the commit for `frontends-backends rules` because, it need to be done once the node is complete functionally.

### CODE REVIEW

**Comment by vkuznet** [comment]
Lina, changes looks fine to me. I understand the logic. But as I wrote in private email we'll need to adjust backend-prod.txt to explicitly add DBS redirect rules to certain nodes (current 4 nodes plus new backend 0766) and have default to current 4 nodes for everybody else.

File: admin/deploy [hunk 1]
```diff
@@ -35,7 +35,7 @@ deploy_admin_post()
 
   local pxlabel pxskip certs
   case $host in
-    vocms0117 | vocms0127 | vocms013[245689] | vocms0734| vocms073[189] | vocms0143 | vocms074[012] | vocms016[01235] | vocms0760 | vocms030[67] | vocms0318 )
+    vocms0117 | vocms0127 | vocms013[245689] | vocms0734| vocms073[189] | vocms0143 | vocms074[012] | vocms016[01235] | vocms0760 | vocms030[67] | vocms0318 | vocms0766)
```
Comment (amaltaro) [inline admin/deploy]: We can match 0760 and 0766 with a regex, so we could have something like (and space is needed before closing parenthesis AFAIK):
```
vocms0117 | vocms0127 | vocms013[245689] | vocms0734 | vocms073[189] | vocms0143 | vocms074[012] | vocms016[01235] | vocms076[06] | vocms030[67] | vocms0318 )
```

File: das/deploy [hunk 2]
```diff
@@ -33,7 +33,7 @@ deploy_das_sw()
 
 deploy_das_post()
 {
-  case $host in vocms013[689] | vocms073[89] | vocms016[135] | vocms0740) disable ;; * ) enable ;; esac
+  case $host in vocms013[689] | vocms073[89] | vocms016[135] | vocms0740 | vocms0766) disable ;; * ) enable ;; esac
```
Comment (amaltaro) [inline das/deploy]: blank space needed here

File: exporters/manage [hunk 3]
```diff
@@ -120,7 +120,7 @@ start()
       ;;
 
     # Apache exporter on production nodes
-    vocms0158 | vocms0760 | vocms0162 | vocms0164 )
+    vocms0158 | vocms0760 | vocms0162 | vocms0164  | vocms0766 )
```
Comment (amaltaro) [inline exporters/manage]: You don't need it here (only frontend nodes)

File: exporters/manage [hunk 4]
```diff
@@ -225,7 +225,7 @@ status()
         process_status "couchdb_exporter"
       ;;
 
-    vocms0158 | vocms0760 | vocms0162 | vocms0164 )
+    vocms0158 | vocms0760 | vocms0162 | vocms0164  | vocms0766 )
```
Comment (amaltaro) [inline exporters/manage]: Same here, please remove.

**Comment by h4d4** [comment]
@bbockelm @yuyiguo @vkuznet Valentin, thanks very much for check it.
Regarding [backends-prod](https://github.com/dmwm/deployment/blob/master/frontend/backends-prod.txt), I need to have it clear from all you experts, because what I understood for @bbockelm commets[0] is that this new gp will also serving the others services as well as the current gps on production, but the update here will be double RAM.
Therefore, to reach above the change will be add `vocms0766` to this line [line](https://github.com/dmwm/deployment/blob/master/frontend/backends-prod.txt#L33).

Brian please let me know your comments.

Best Regards Lina.
[0] Brian's comments: 
"I'm suggesting upgrading one (so we have N+1 capacity), then sequentially touching the others.  This way, we always maintain N DBS backends and never N-1 backends.  We should upgrade the memory on all of them while we're at it."

**Comment by amaltaro** [comment]
I'm not sure I got what Valentin wanted to say, but I think he said the same as Brian and the same as everyone understood, for the moment.
We add a new node to CMSWEB - which can serve any GP requests, thus running any GP services - then the other nodes start going down one by one, which means we will always have 4 GP backends up & running.

Having a specific backend for a specific client is out of the scope for this very near future, AFAICT.

**Comment by vkuznet** [comment]
Lina,
yes, the change would be to add `vocms0766` to last line.

File: admin/deploy (hunk 1 above)
Comment (h4d4) [reply admin/deploy]: Yes, it true.

File: das/deploy (hunk 2 above)
Comment (h4d4) [reply das/deploy]: the same, I'll add the space.

File: exporters/manage (hunk 3 above)
Comment (h4d4) [reply exporters/manage]: yes, thanks! it is a mistaken.

File: exporters/manage (hunk 4 above)
Comment (h4d4) [reply exporters/manage]: yes, the same error.

**Comment by h4d4** [comment]
@amaltaro Alan,

thanks for the review, I'll apply the changes and let you know. 

Best Regards, Lina.

**Comment by h4d4** [comment]
@amaltaro Alan, 
I've just applied above pointed  corrections.

**Comment by amaltaro** [comment]
It looks good now, Lina. However, if it was a PR against WMCore, I'd never merge it because it has 9 commits! Given it's under the deployment repo, you do as it pleases you :), or you spend < 5 and squash all those commits in a single one, anyone looking at the history of those files will be happy to see a single commit instead of commits back and forth.

**Comment by h4d4** [comment]
@amaltaro Alan, 
thanks for the comments I'll keep in mind it to improve :) 
Yes since it's deployment repo I'm responsible to merge it.
I'll review git doc for squashing 

Best Regards, Lina.

**Comment by amaltaro** [comment]
@h4d4 just in case you missed this PR, I believe you wanted to push it to testbed in HG1906, no?

**Comment by h4d4** [comment]
@amaltaro Alan, 

Thanks for check it. nevertheless, this PR was overseed by PR #766.
I'll close this one.

**Comment by h4d4** [comment]
Closing PR, since it was overseed by PR #766.

**Pull request closed without merging**